### ISSUE TITLE
Possible issue with typings where the parameters can be `null`.

### ISSUE THREAD

**Issue by TheBox193** [JaroslawPokropinski/eslint-plugin-strict-null-checks#21]
Seems that the checker is not picking up on a libraries typings where the parameters can indeed be `null`.

```
Don't pass nullish arguments to a function that expects a non-nullish type.eslint[strict-null-checks/all](file:///all)
```

![image](https://user-images.githubusercontent.com/220755/217969780-549f67c1-8ede-403f-9285-f76c90f2121b.png)

![image](https://user-images.githubusercontent.com/220755/217969783-ecf39652-888a-464e-b9cf-e201c591f6a9.png)

**Comment by JaroslawPokropinski** [comment]
Are non-imported functions working correctly? For example does adding this not generate a warning?:
```typescript
function foo(value: {value: number} | null) {}
foo(null);
```

**Comment by doytch** [comment]
First off, thanks for your work on this plugin @JaroslawPokropinski. 

Just ran into this as well with a slightly different usage (function param initializers), and I can confirm it doesn't work for non-imported functions. Here's a trivial example:
<img width="589" alt="Screen Shot 2023-02-23 at 15 30 53" src="https://user-images.githubusercontent.com/5324374/221023282-e98ee2ba-0e7c-4805-ad28-3b337f3bf1c1.png">

In that example, TS tweaks the type of `foo` from what we defined it as to `const foo: (bar?: number) => number` because of the `bar = 2` initializer.

**Comment by neolectron** [comment]
Hello @doytch, have you been able to find the root cause of this issue ?

I'm also have this other issue (that look familiar).

![image](https://github.com/JaroslawPokropinski/eslint-plugin-strict-null-checks/assets/5268400/ae866a54-9441-4dce-b8ce-87fef13c759e)

**Comment by doytch** [comment]
That’s an odd one. But no, I never found a solution. 

What I ended up doing is stopping use of this plugin in our org and used an interesting little feature in VSCode instead since almost all of us use it. 

I added a nested tsconfig in our codebase. That is, if our normal tsconfig was at the project root ~, I put a new tsconfig in ~/src. 

That tsconfig simply inherited the main tsconfig but turned on strictNullChecks. Absolutely nothing uses a nested tsconfig file by default - except VSCode’s TS server.

So that showed everybody errors during development but didn’t break builds. That may or may not meet your requirements (and I thiiink we might’ve been relying on undocumented VSCode behaviour so it’s not reliable long-term), but just figured I’d throw it out there.
